vault/src/process.rs

```diff
@@ -0,0 +1,230 @@
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::PathBuf,
+    process,
+    str::FromStr,
+};
+
+use runtime::AccountId;
+use service::Error;
+use sysinfo::{Pid, PidExt, ProcessExt, System, SystemExt};
+
+pub trait SystemProcess {
+    fn refresh_process(&mut self, pid: Pid) -> bool;
+    fn process_name(&self, pid: Pid) -> Result<String, Error>;
+}
+
+impl SystemProcess for System {
+    fn refresh_process(&mut self, pid: Pid) -> bool {
+        <Self as SystemExt>::refresh_process(self, pid)
+    }
+
+    fn process_name(&self, pid: Pid) -> Result<String, Error> {
+        let process = <Self as SystemExt>::process(self, pid).ok_or_else(|| Error::ProcessNotFound(pid.to_string()))?;
+        Ok(process.name().to_string())
+    }
+}
+
+#[derive(Debug)]
+pub struct PidFile {
+    path: PathBuf,
+}
+
+impl PidFile {
+    pub fn new(spec_name: &String, account_id: &AccountId) -> Self {
+        Self {
+            path: Self::compute_path(spec_name, account_id),
+        }
+    }
+
+    pub fn create(spec_name: &String, account_id: &AccountId, sys: &mut impl SystemProcess) -> Result<Self, Error> {
+        let pid_file = Self::new(spec_name, account_id);
+        if pid_file.path.exists() {
+            let pid = pid_file.pid()?;
+            if sys.refresh_process(pid) {
+                match pid_name_matches_existing_client(sys, pid) {
+                    Ok(true) => return Err(Error::ServiceAlreadyRunning(pid.to_string())),
```

Comment:
Why convert to a string?

Reply:
Yea makes sense, just wondering if we should use the u32 instead (assuming that's what `Pid` uses internally).